src/renderer/components/Tables/VideoSelectionTable.tsx

```diff
@@ -0,0 +1,569 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { AppState, RendererVideo } from 'main/types';
+
+import {
+  Cell,
+  ColumnDef,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getSortedRowModel,
+  Header,
+  Row,
+  useReactTable,
+} from '@tanstack/react-table';
+import {
+  Fragment,
+  MutableRefObject,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import ViewpointSelection from 'renderer/components/Viewpoints/ViewpointSelection';
+import ViewpointInfo from 'renderer/components/Viewpoints/ViewpointInfo';
+import ViewpointButtons from 'renderer/components/Viewpoints/ViewpointButtons';
+import StateManager from 'renderer/StateManager';
+import RaidEncounterInfo from 'renderer/RaidEncounterInfo';
+import RaidCompAndResult from 'renderer/RaidComp';
+import { VideoCategory } from 'types/VideoCategory';
+import DungeonInfo from 'renderer/DungeonInfo';
+import { ArrowDown, ArrowUp } from 'lucide-react';
+import {
+  countUniqueViewpoints,
+  getDungeonName,
+  getInstanceDifficultyText,
+  getPullNumber,
+  povNameSort,
+  videoToDate,
+} from '../../rendererutils';
+import {
+  DateHeader,
+  DifficultyHeader,
+  DurationHeader,
+  EncounterHeader,
+  LevelHeader,
+  MapHeader,
+  PullHeader,
+  ResultHeader,
+  TagHeader,
+  TypeHeader,
+  ViewpointsHeader,
+} from './Headers';
+import { durationSort, levelSort, resultSort } from './Sorting';
+import {
+  populateDateCell,
+  populateDetailsCell,
+  populateDurationCell,
+  populateLevelCell,
+  populateResultCell,
+} from './Cells';
+
+interface IProps {
+  videoState: RendererVideo[];
+  appState: AppState;
+  setAppState: React.Dispatch<React.SetStateAction<AppState>>;
+  stateManager: MutableRefObject<StateManager>;
+  persistentProgress: MutableRefObject<number>;
+}
+
+/**
+ * Table component for displaying available videos. Includes category appropriate
+ * columns for a quick overview, the ability to sort by column, and the option to
+ * expand a specific activity for more details and controls.
+ */
+const VideoSelectionTable = (props: IProps) => {
+  const {
+    videoState,
+    appState,
+    setAppState,
+    stateManager,
+    persistentProgress,
+  } = props;
+
+  const [expanded, setExpanded] = useState<ExpandedState>({});
+
+  /**
+   * Reset expanded on changing category.
+   */
+  useEffect(() => {
```

Comment:
seems like this would be better set in the select category handler, higher in the stack

Reply:
I left a comment. Not changing for now.